tt_metal/python_env/module.mk

```diff
@@ -30,10 +34,18 @@ endif
 	bash -c "source $(PYTHON_ENV)/bin/activate && python3 -m pip install -r tt_metal/python_env/requirements-dev.txt"
 	echo "Installing editable dev version of tt_eager packages..."
 	bash -c "source $(PYTHON_ENV)/bin/activate && pip install -e ."
+	touch $@
+
+$(PYTHON_ENV)/.installed-dev-editable:
```

Comment:
Hmm... the editable step doesn't include `metal-libs` editable install? I'm assuming this is because we need the dependencies installed through `pip install -e .` in the root dir to actually get the dependencies we need to cache into the cached environment.

I'm wondering if we could try refactoring this section into:

```
.PHONY: python_env python_env/dev python_env/dev/editable python_env/dev/stubs python_env/clean

python_env/dev: python_env tt_metal/python_env/requirements-dev.txt
	echo "Installing dev environment packages..."
	bash -c "source $(PYTHON_ENV)/bin/activate && python3 -m pip install -r tt_metal/python_env/requirements-dev.txt"

python_env/dev/editable: python_env/dev $(TT_LIB_LIB_LOCAL_SO) $(TTNN_PYBIND11_LOCAL_SO)
	echo "Installing editable dev version of tt_eager packages..."
	bash -c "source $(PYTHON_ENV)/bin/activate && pip install -e ."
	echo "Installing editable dev version of ttnn package..."
	bash -c "source $(PYTHON_ENV)/bin/activate && pip install -e ttnn"

python_env/dev/stubs: 
    echo "Generating stubs..."
    ...
```

Let me know you're thoughts. I know that was a lot of mumbo jumbo I just spewed

Reply:
Let me rework this section and you and I can re-review it together. 
`Editable` was meant to depend on `python/dev`

Reply:
There are a few issues in the proposed code that make me believe we are not on the same page.
Let me know when you are free to discuss and we can review and chat through this together.

Specifically:

- Installing `pip install -e ."` as part of editable and not cached requirements is going to miss a lot of dependencies. I would like to cache the dependencies in pyproject.toml as well. Maybe the name of the target `editable` is what is confusing.  Maybe we can change to `pip install .`
- `python_env/dev/stubs` needs `python_env/dev/editable` but if they are all .PHONY then it will get rebuilt causing a 5 min delay per test.